benchmark/cluster_containers_setup.py

```diff
@@ -208,10 +318,36 @@ def deploy(self):
         Message.log(action="Listing current nodes")
         d = self.client.list_nodes()
         d.addCallback(self._set_nodes)
-        d.addCallback(self._build_config)
-        d.addCallback(self._configure)
+        d.addCallback(self._set_current_number_of_datasets_and_containers)
+        d.addCallback(self.create_datasets_and_containers)
         return d
 
+    def _set_current_number_of_datasets_and_containers(self, ignored):
+        """
+        Populates the ``self._initial_num_containers`` with the current
+        number of containers in the cluster. It is intended to be used
+        before requesting the creation of any containers or datasets so
+        we can know the total number of datasets and containers to expect.
+
+        :return Deferred: that will fire once the ``_initial_num_datasets``
+            and ``_initial_num_containers`` are populated.
+        """
+        d1 = self.client.list_containers_state()
+
+        def set_initial_num_containers(containers):
+            self._initial_num_containers = len(containers)
```

Comment:
:question: These are still unnecessary state, but again not worth doing more work on.